tests/unit/test_server.py

```diff
@@ -9,42 +9,84 @@
 
 
 class TestServer(unittest.TestCase):
-    def setUp(self):
-        self.prometheus = Prometheus("localhost", 9090)
-
-    @responses.activate
-    def test_prometheus_server_returns_valid_data(self):
-        version = "1.0.0"
-
-        responses.add(
-            responses.GET,
-            "http://localhost:9090/api/v1/status/buildinfo",
-            json={
-                "status": "success",
-                "data": {"version": version},
-            },
-            status=200,
-        )
-
-        got_version = self.prometheus.version()
-        self.assertEqual(got_version, version)
-
-    @responses.activate
-    def test_prometheus_server_reload_configuration_success(self):
-        responses.add(
-            responses.POST,
-            "http://localhost:9090/-/reload",
-            status=200,
-        )
-
-        self.assertTrue(self.prometheus.reload_configuration())
-
-    @responses.activate
-    def test_prometheus_server_reload_configuration_failure(self):
-        responses.add(
-            responses.POST,
-            "http://localhost:9090/-/reload",
-            status=500,
-        )
-
-        self.assertFalse(self.prometheus.reload_configuration())
+    class _TestServerWithoutRoutePrefix(unittest.TestCase):
```

Comment:
None of these tests ran: https://github.com/canonical/prometheus-k8s-operator/runs/5485103377?check_suite_focus=true#step:5:1

I don't think nested test classes are picked up automatically.
But why nest them?

Reply:
Crud you are right